pkg/client/core/api/rest/client.go

```diff
@@ -104,3 +116,22 @@ func (c *HTTPClient) Do(method, endpoint string, body interface{}, into interfac
 func pretty(msg, method, endpoint string) string {
 	return fmt.Sprintf("%s: %s, %s", msg, method, endpoint)
 }
+
+type serializedError struct {
+	Detail map[string]interface{} `json:"detail,omitempty"`
+	Type   string                 `json:"type"`
+	Error  string                 `json:"error"`
+	Code   errors.Kind            `json:"code"`
+}
+
+// jsonAsMishudarkError converts an error from the backend to a mishudark error
+func jsonAsMishudarkError(jsonContent []byte, defaultError error) error {
+	data := serializedError{}
+
+	err := json.Unmarshal(jsonContent, &data)
+	if err != nil {
+		return defaultError
```

Comment:
The go JSON marshaller isn't strict, so if something doesn't match the struct you pass in, it will not fail. It will generally only fail if we are not unmarshalling proper JSON (https://stackoverflow.com/questions/19633763/unmarshaling-json-in-golang-required-field)

Reply:
So in other words, create an internal error instead only complaining about unmarshalling being given bad content?

Reply:
In the example I posted earlier I always decode into the structure below if there is an error. Basically, we need to guarantee that any error that is returned by the server conforms to some "schema" and then we need to unmarshal into that schema

```go
type ErrTimeout struct {
	Message string
}

func (e *ErrTimeout) Error() string {
	return e.Message
}

type Error struct {
	Type   string  `json:"type"`
	Error  string  `json:"error"`
	Code   int     `json:"code"`
	Detail *Detail `json:"detail,omitempty"`
}

type Detail struct {
	Timeout *ErrTimeout `json:"timeout,omitempty"`
}
```

Reply:
I misunderstood your code! Just disregard me